src/unit_tests/kubernetes_api_objects/container.rs

```diff
@@ -0,0 +1,357 @@
+    println!("Testing set_name() for volume mount...");
+    let mut volume_mount = VolumeMount::default();
+    volume_mount.set_name(new_strlit("name").to_string());
+    assert_eq!("name".to_string(), volume_mount.into_kube().name);
+}
+
+#[test]
+#[verifier(external)]
+pub fn test_set_read_only() {
+    println!("Testing read_only()...");
+    let mut volume_mount = VolumeMount::default();
+    volume_mount.set_read_only(true);
+    assert_eq!(true, volume_mount.into_kube().read_only.unwrap());
+}
+
+#[test]
+#[verifier(external)]
+pub fn test_set_sub_path() {
+    println!("Testing sub_path()...");
+    let mut volume_mount = VolumeMount::default();
+    volume_mount.set_sub_path(new_strlit("sub_path").to_string());
+    assert_eq!("sub_path".to_string(), volume_mount.into_kube().sub_path.unwrap());
+}
+
+#[test]
+#[verifier(external)]
+pub fn test_set_volume_mounts() {
+    println!("Testing set_volume_mounts()...");
+    let mut container = Container::default();
+    let volume_mounts = || {
+        let mut volume_mount = VolumeMount::default();
+        volume_mount.set_mount_path(new_strlit("mount_path").to_string());
+        volume_mount.set_read_only(true);
+        volume_mount.set_sub_path(new_strlit("sub_path").to_string());
+        let mut volume_mounts = Vec::new();
+        volume_mounts.push(volume_mount);
+        volume_mounts
+    };
+    container.set_volume_mounts(volume_mounts());
+    assert_eq!(volume_mounts().into_iter().map(|v: VolumeMount| v.into_kube()).collect::<Vec<_>>(),
+               container.into_kube().volume_mounts.unwrap());
+}
+
+// Tests for ports(ContainerPort)
+#[test]
+#[verifier(external)]
+pub fn test_set_container_port() {
+    println!("Testing set_container_port()...");
+    let mut container_port = ContainerPort::default();
+    container_port.set_container_port(8080);
+    assert_eq!(8080, container_port.into_kube().container_port);
+}
+
+#[test]
+#[verifier(external)]
+pub fn test_set_name_container_port() {
+    println!("Testing set_name() for container port...");
+    let mut container_port = ContainerPort::default();
+    container_port.set_name(new_strlit("name").to_string());
+    assert_eq!("name".to_string(), container_port.into_kube().name.unwrap());
+}
+
+#[test]
+#[verifier(external)]
+pub fn test_set_ports() {
+    println!("Testing set_ports()...");
+    let mut container = Container::default();
+    let ports = || {
+        let mut container_port = ContainerPort::default();
+        container_port.set_container_port(8080);
+        container_port.set_name(new_strlit("name").to_string());
+        let mut ports = Vec::new();
+        ports.push(container_port);
+        ports
+    };
+    container.set_ports(ports());
+    assert_eq!(ports().into_iter().map(|v: ContainerPort| v.into_kube()).collect::<Vec<_>>(),
+               container.into_kube().ports.unwrap());
+}
+
+// Tests for life cycle
+#[test]
+#[verifier(external)]
+pub fn test_set_command_execaction() {
```

Comment:
Just call it set_command

Reply:
There are two set_command need to be tested, so I add a tail for this one. The other one is set_command for container.

Reply:
Then you can use different mod to separate tests for different structs

Reply:
I just tried to keep the same structure as our kubernetes_api_objects directory. Wouldn't keeping the same layer/structure make the tests easier to be found/read?

Reply:
No I am not asking you to divide them into different files. Keep them in the same file, but different mod. See http://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/book/second-edition/ch07-01-mod-and-the-filesystem.html#module-definitions -- you will have one module for one struct you are testing. Also see my comment before https://github.com/vmware-research/verifiable-controllers/pull/278#discussion_r1328972589

Reply:
Sure, I will fix it.